test-cypress/integration/repository/ontop-repository.spec.js

```diff
@@ -152,6 +239,131 @@ describe('Ontop repositories', () => {
+        cy.waitUntil(() =>
+            RepositorySteps.getRepositoryFromList(repositoryId)
+                .should('be.visible')
+                .find('.repository-status .text-secondary').should('contain.text', status));
+    }
+
+    const commonRepositoryParams = {
+        "propertiesFile": {
+            "name": "propertiesFile",
+            "label": "JDBC properties file",
+            "value": "repositories/ontop/ontop_jdbc.properties"
+        },
+        "lensesFile": {
+            "name": "lensesFile",
+            "label": "Lenses file",
+            "value": ""
+        },
+        "isShacl": {
+            "name": "isShacl",
+            "label": "Enable SHACL validation",
+            "value": "false"
+        },
+        "owlFile": {
+            "name": "owlFile",
+            "label": "Ontology file",
+            "value": ""
+        },
+        "member": {
+            "name": "member",
+            "label": "FedX repo members",
+            "value": []
+        },
+        "constraintFile": {
+            "name": "constraintFile",
+            "label": "Constraint file",
+            "value": ""
+        },
+        "obdaFile": {
+            "name": "obdaFile",
+            "label": "OBDA or R2RML file",
+            "value": "repositories/ontop/university-complete.obda"
+        },
+        "dbMetadataFile": {
+            "name": "dbMetadataFile",
+            "label": "DB metadata file",
+            "value": ""
+        }
+    };
+
+    function interceptRepository(id, statusCode, extraParams = {}) {
+        cy.intercept(`rest/repositories/${id}?location=`, {
+            statusCode,
+            body: {
+                "id": id,
+                "title": "",
+                "type": "ontop",
+                "sesameType": "graphdb:OntopRepository",
+                "location": "",
+                "hostName": "localhost",
+                "params": { ...commonRepositoryParams, ...extraParams }
+            }
+        });
+    }
+
+    function stubRepoCreationEndpoints() {
+        cy.intercept(REPOSITORIES_URL + 'all', {
+            statusCode: 200,
+            body: {
+                "": [
+                    {
+                        "id": repositoryId,
+                        "title": "",
+                        "uri": "http://address:5423/repositories/ontop",
+                        "externalUrl": "http://address:5423/repositories/ontop",
+                        "local": true,
+                        "type": "ontop",
+                        "sesameType": "graphdb:OntopRepository",
+                        "location": "",
+                        "readable": true,
+                        "writable": true,
+                        "unsupported": false,
+                        "state": "ACTIVE"
+                    }
+                ]
+            }
+        }).as('getMockRepositories');
+
+        cy.intercept(`rest/repositories/${repositoryId}/restart?location=*`, {
```

Comment:
Also try to implement the stubs in their respective classes so that we can reuse and compose them.